### ISSUE TITLE
Add support for spark 2.2.x

### ISSUE THREAD

**Issue by AliTajeldin** [TresAmigosSD/SMV#933]
spark 2.2.0 (https://spark.apache.org/releases/spark-release-2-2-0.html) is out.  We should wait until 2.2.1 is out to add support for SMV as ".0" releases of spark tend to be rather buggy.

**Comment by laneb** [comment]
Closing as duplicate of #1120.

**Comment by cloudysail** [comment]
I suggest we can have this ticket to upgrade our 2.2.0 to 2.2.1

**Comment by laneb** [comment]
What would be the difference between this ticket and 2.2.1?

**Comment by cloudysail** [comment]
Right now our SMV 2.2 is using spark 2.2.0. I tried to upgrade to use Spark 2.2.1 a few months ago but was blocked by some compilation error. Maybe we should continue to effort to resolve the spark upgrade issue

**Comment by cloudysail** [comment]
Oh, sure. Then we trace it in #1120
